libp2p/stream/lpstream.nim

```diff
@@ -7,20 +7,21 @@
 ## This file may not be copied, modified, or distributed except according to
 ## those terms.
 
-import chronicles, chronos
+import oids
+import chronicles, chronos, metrics
 import ../varint,
-       ../vbuffer
+       ../vbuffer,
+       ../peerinfo,
+       ../multiaddress
 
-when chronicles.enabledLogLevel == LogLevel.TRACE:
-  import oids
+declareGauge(libp2p_open_streams, "open stream instances", labels = ["type"])
 
 type
   LPStream* = ref object of RootObj
     isClosed*: bool
     isEof*: bool
-    closeEvent*: AsyncEvent
-    when chronicles.enabledLogLevel == LogLevel.TRACE:
-      oid*: Oid
+    objName*: string
+    oid*: Oid
```

Comment:
still not sure why we need `oid` vs `cast[int](self)` since its a `ref`

Reply:
I was actually thinking about it, probably better to just use `hash()` either way, whatever it ends up being under the hood.

Reply:
and probably a good idea to have a `hash()` implemented for most objects either way

Reply:
Yes hash will use the pointer just like the cast iirc!

Reply:
I'll rework it in the next pr, don't want this one to go stale...